### DISCUSSION TITLE
JSONPath: Replacing result nodes in data?

### DISCUSSION THREAD

**Discussion by kabukky** [ohler55/ojg#94] [Q&A] [answered]
Hi,

this library seems pretty powerful.
I'm hoping to achive the following, but couldn't find a good way yet. So I'm hoping for a pointer on how to go about it:

I'd like to query data from a JSON object via a JSONPath expression, alter the found fields in-place, and write it all back into a JSON object.

E. g.:

```
package main

import (
	"log"

	"github.com/ohler55/ojg/alt"
	"github.com/ohler55/ojg/jp"
	"github.com/ohler55/ojg/oj"
)

var (
	input = oj.MustParseString(`{
		"a":[
			{"x":1,"y":2,"z":3},
			{"x":2,"y":4,"z":6}
		]
	}`)
	newData = oj.MustParseString(`{
		"foo": "bar"
	}`)
)

func main() {
	inputNode := alt.GenAlter(input)
	expression, err := jp.ParseString("a[?(@.x > 1)].y")
	if err != nil {
		log.Fatalln(err)
	}
	nodes := expression.GetNodes(inputNode)
	for _, node := range nodes {
		// Set doesn't exist
		// but I'd like to replace the data inside the node to alter it in-place
		node.Set(alt.GenAlter(newData))
	}
	log.Println(inputNode)
	//  Want:
	// {"a":[{"x":1,"y":2,"z":3},{"y":{"foo":"bar"},"z":6,"x":2}]}
}
```

This is just an example to hopefully illustrate what I want to do. I'm not sure if Nodes are the right way to go about it. Do you have a pointer for me on how to go about this?

Thanks

**Comment by ohler55** [comment]
It's actually much simpler than you think. I'll provide something this evening after work.

**Comment by ohler55** [comment, marked as answer]
Keeping the imports and vars as they are just change main() to:

```go
func main() {
	x := jp.MustParseString("a[?(@.x > 1)].y")
	x.MustSet(input, newData)
	fmt.Println(pretty.JSON(input))
}
```

outputs:

```json
{
  "a": [
    {"x": 1, "y": 2, "z": 3},
    {"x": 2, "y": {"foo": "bar"}, "z": 6}
  ]
}
```

**Reply by kabukky** [reply]
Thanks for anwering my question! That solves the posed problem.

But I just now realized that my question was too generalized. I'll open a new one discussion so this doesn't get too confusing.
